### PR TITLE
Set device type to BackEnd if enable_compute_ai_deployment is true

### PR DESCRIPTION
**Pull request by wsycqyz**
### Description of PR

Summary:
Set device type to BackEndToRRouter or BackEndLeafRouter if enable_compute_ai_deployment is true during minigraph .xml generation.
sonic-mgmt-internal branch PR#6352

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
Set device type to BackEndToRRouter or BackEndLeafRouter if enable_compute_ai_deployment is true during minigraph .xml generation.

#### How did you verify/test it?
Verify with physical testbed internally.
image

### CODE REVIEW

**Comment by kevinskwang** [comment]
@wsycqyz could you change the neighbor's type as well?

File: ansible/templates/minigraph_png.j2 [hunk 1]
```diff
@@ -113,7 +113,17 @@
 {% endif %}
     </DeviceInterfaceLinks>
     <Devices>
+{% if enable_compute_ai_deployment|default('false')|bool %}
```
Comment (XuChen-MSFT) [inline ansible/templates/minigraph_png.j2:116]: saw most cases in this repo using below syntax for checking bool value
{% if (enable_compute_ai_deployment is defined) and (enable_compute_ai_deployment == True) %}

wondering if your syntax pass the test?
{% if enable_compute_ai_deployment|default('false')|bool %}

Comment (wsycqyz) [reply ansible/templates/minigraph_png.j2:116]: If the enable_compute_ai_deployment variable is not defined or its value is empty, the default value 'false' will be used.

**Review by XuChen-MSFT — APPROVED** [review]
Good to go!

**Review by kevinskwang — APPROVED** [review]

**Pull request merged**